Web.md

```diff
@@ -20,6 +20,44 @@ WebAssembly's [modules](Modules.md) allow for natural [integration with
 the ES6 module system](Modules.md#integration-with-es6-modules) and allow
 synchronous calling to and from JavaScript.
 
+### Function Names
+
+A WebAssembly module imports and exports functions. WebAssembly names functions
+using arbitrary-length byte sequences. Any 8-bit values are permitted in a
+WebAssembly name, including the null byte and byte sequences that don't
+correspond to any Unicode code point regardless of encoding. The most natural
+Web representation of a mapping of function names to functions is a JS object
+in which each function is a property. Property names in JS are UTF-16 encoded
+strings. A WebAssembly module may fail validation on the Web if it imports or
+exports functions whose names do not transcode cleanly to UTF-16 according to
+the following conversion algorithm, assuming that the WebAssembly name is in a
+`Uint8Array` called `array`:
+
+```
+function convertToJSString(array)
+{
+  // Perform the actual conversion.
+  var string = "";
+  for (var i = 0; i < array.length; ++i)
+    string += String.fromCharCode(array[i]);
+  var result = decodeURIComponent(escape(string));
+  
+  // Check for errors. This will throw if 'result' contains bad characters.
+  encodeURIComponent(result);
```

Comment:
Ah hah, and poking around more, it would appear that it's simplify invalid utf8 to store a code point in the surrogate range \[0xd800, 0xdc00\] (I had been thinking that utf8 wouldn't care about surrogates, but I guess this is a concession to utf16 baked into utf8?), so your definition here is *precisely* what we want (if we want to reuse our existing Utf8ToUtf16 conversion routines which I very much do.  lgtm!